### PR TITLE
Ui: eslint custody

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?

- pulls UI's own eslint into line with frontend
- makes linting dirs optin, rather than universal, for perf reasons
- updates the validate task to lint UI
- runs more validate tasks in parallel 
  screen shot 2017-08-09 at 15 25 43

## What is the value of this and can you measure success?

i was going to detach UI's listing from the rest of frontend entirely, but isolating it became complicated when considering things like prettier. 

do we want the same settings? probably. in which case, why duplicate it? why have to maintain 2 identical instances of eslint? why not make the whole expect the same settings where applicable? 

this treats UI like we do legacy and the main app. a simplified `eslintrc` manages stuff like prettier, while apps get more complex and bespoke.

## Does this affect other platforms - Amp, Apps, etc?

shoudn't do

## Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

i have no idea – presumably not?

### CODE REVIEW

**Review by gu-stav — APPROVED** [review]
I agree with your points made in the description. It doesn't make it harder to extract it at some point, because we could then still duplicate the configs.

On the other hand it makes it easier to maintain a consistent style across all the JS files.

File: dev/.eslintrc.js [hunk 1]
```diff
@@ -1,7 +1,9 @@
 module.exports = {
+    env: {
+        browser: true,
+    },
```
Comment (SiAdcock) [inline dev/.eslintrc.js]: There's browser code in `dev/`?

**Comment by SiAdcock** [comment]
As a devil's advocate argument, we could extract our ESLint config (and maybe our prettier config?) into an installable config (like AirBnB have done). That would save maintaining multiple instances, while promoting consistency and keeping `ui/` extractable.

Maybe we can think about whether we want to do this if we choose to extract `ui/` in the future 😄

**Comment by sndrs** [comment]
@SiAdcock that is a very nice idea actually. gets rid of inheritance too, which is not really very transparent

File: dev/.eslintrc.js (hunk 1 above)
Comment (sndrs) [inline dev/.eslintrc.js]: yeah in `dev/jest.setupTestFrameworkScriptFile.js`

**Comment by sndrs** [comment]
i'm merge this but we should look at that. i guess this gives us a starting point to do that

**Pull request merged**